types/aws-lambda/aws-lambda-tests.ts

```diff
@@ -899,3 +899,47 @@ let SQSMessageNode8AsyncHandler: AWSLambda.SQSHandler = async (
     str = context.functionName;
     return;
 };
+
+const FirehoseNode8AsyncEventHandler: AWSLambda.FirehoseTransformationHandler = async (
+    event: AWSLambda.FirehoseEvent,
+    context: AWSLambda.Context
+) => {
+    // $ExpectType FirehoseEvent
+    event;
+    str = event.records[0].recordId;
+
+    // $ExpectType Context
+    context;
+    str = context.functionName;
+    return {
+        records: [
+            {
+                recordId: str,
+                result: 'Ok' as AWSLambda.FirehoseRecordTransformationStatus,
+                data: 'eyJmb28iOiJiYXIifQ==',
+            }
+        ]
+    };
+};
+const FirehoseEventHandler: AWSLambda.FirehoseTransformationHandler = (
+    event: AWSLambda.FirehoseEvent,
+    context: AWSLambda.Context,
+    callback: AWSLambda.Callback
+) => {
+    // $ExpectType FirehoseEvent
+    event;
+    str = event.records[0].recordId;
+
+    // $ExpectType Context
+    context;
+    str = context.functionName;
+    callback(null, {
+        records: [
+            {
+                recordId: event.records[0].recordId,
+                result: 'Ok' as AWSLambda.FirehoseRecordTransformationStatus,
+                data: 'eyJmb28iOiJiYXIifQ==',
+            }
+        ]
+    });
+};
```

Comment:
These handler tests are overkill - which isn't bad except for making the next reviewer think they need to do this much, like the last person did to you 😅. I didn't block them for it, so I won't for you either.

If you want to clean this up a bit, the second one that tests the callback type is the one to keep, and the naming convention for the handler would be the camel case of the handler type name, but those a pretty trivial details, the rest of this file has bigger issues!

(start of me babbling out loud:)

Really this whole file should probably be sorted out, and I do like the test looking like a real handler, rather than the kind of dumb looking `str = someEvent.someStringField` tests above, but that would be a separate PR (maybe I'll get around to it some time).

There's a few different tests for the first few handler definitions because it`s testing the definition for `Handler<>`, which is a bit weird of a type, the other one-line handlers are just testing that the `FooHandler` is compatible with `FooEvent` and `FooCallback` to catch copy-paste errors.

Reply:
I removed the async one and adjusted for the specialized callback. I did notice you added the customHandler and no types on the 3 arguments and then you expect on the types. Is there a way to reuse that or I need to replicate the same tests as there?

Reply:
I broke the tests after forgetting to change the expectations. We are often using a package called [husky](https://www.npmjs.com/package/husky) to run tests and lints pre git push so we don't have any surprises when the CI tool runs. Maybe it's a good thing to add? Or you think it adds too much overhead on everyone contributing?

Reply:
The custom handler tests are checking using the handler type parameters doesn't break, while the FooHandlers are checking that the handler and callback types are correctly defined. There's no real reason to expect types, since they're explicitly set just above. On the other hand, checking that you can read the event properties and write the result as you would expect adds value, that's what all the weird assignments above the handlers are doing, and the bit I was referring to being a cleanup I'd like to do at some point. I wouldn't be too worried about it, myself, since these tests really are supposed to be code that can actually run, which obviously is pretty tricky for environment packages like this.

On husky, not sure, might be worth asking in a meta issue if the maintainer that merges this doesn't hate it. I'm kinda neutral myself, as the only person that really gets burned is the person that pushed without testing, and the CI is pretty fast here.